src/freenet/node/NewPacketFormat.java

```diff
@@ -10,6 +10,7 @@
 import java.util.Iterator;
 import java.util.LinkedList;
 import java.util.List;
+import java.util.concurrent.ConcurrentSkipListMap;
```

Comment:
Improving upon what? `ConcurrentSkipListMap` iteration may be faster but at increased memory cost. Maybe a `LinkedHashMap` would suffice?

(Also, I’m a fan of interfaces in type declarations so I’d prefer having `startedByPrio` declared as `List<Map<Integer, MessageWrapper>>`…)